cmd/client-s3.go

```diff
@@ -795,15 +805,47 @@ func (c *S3Client) Watch(ctx context.Context, options WatchOptions) (*WatchObjec
 	return wo, nil
 }
 
+type ttfbRecorder struct {
+	readStart time.Time
+	source    string
+	rc        io.ReadCloser
+}
+
+func (f *ttfbRecorder) Read(p []byte) (n int, err error) {
+	if !f.readStart.IsZero() || len(p) == 0 {
+		return f.rc.Read(p)
+	}
+
+	f.readStart = time.Now()
+	n, err = f.rc.Read(p)
+	if n > 0 {
+		mcS3GETLatency.With(prometheus.Labels{"endpoint": f.source}).Set(float64(time.Since(f.readStart)))
+	} else {
+		f.readStart = time.Time{}
+	}
```

Comment:
wouldn't this cause first latency to be wrong? `time.Since(f.readStart)` and readStart is time.Time{}

Reply:
> wouldn't this cause first latency to be wrong? `time.Since(f.readStart)` and readStart is time.Time{}

Simplifed it further.